clang-tools-extra/clang-tidy/ClangTidyDiagnosticConsumer.cpp

```diff
@@ -311,7 +311,12 @@ ClangTidyDiagnosticConsumer::ClangTidyDiagnosticConsumer(
     : Context(Ctx), ExternalDiagEngine(ExternalDiagEngine),
       RemoveIncompatibleErrors(RemoveIncompatibleErrors),
       GetFixesFromNotes(GetFixesFromNotes),
-      EnableNolintBlocks(EnableNolintBlocks) {}
+      EnableNolintBlocks(EnableNolintBlocks) {
+
+  if (Context.getOptions().ExcludeHeaderFilterRegex)
+    ExcludeHeaderFilter = std::make_unique<llvm::Regex>(
+        *Context.getOptions().ExcludeHeaderFilterRegex);
+}
```

Comment:
It's not a good idea to have basically the same thing implemented in two different ways (`HeaderFilter` vs `ExcludeHeaderFilter`). I think the `ExcludeHeaderFilter` should be implemented like the `HeaderFilter`.
https://github.com/llvm/llvm-project/blob/dcf92a249233cab103f848dd12e96e0d642a8899/clang-tools-extra/clang-tidy/ClangTidyDiagnosticConsumer.cpp#L574-L579
This would also solve @PiotrZSL's comment.

We can probably save the constructions of `llvm::Regex` and the call to `match` on the empty regex. wdyt? This could also be done in another pr.

Reply:
Well, in the initial PR `ExcludeHeaderFilter` mirrored `HeaderFilter` behavior exactly. I changed it because @PiotrZSL made a nearly identical comment to yours about saving the call to `match` on an empty regex (which I think is handled in the latest revision).

I _could_ migrate `HeaderFilter` to mirror the new and improved `ExcludeHeaderFilter`...though I worry a bit about this change growing larger and larger. What do you think?

Reply:
> which I think is handled in the latest revision

I agree

> I could migrate HeaderFilter to mirror the new and improved ExcludeHeaderFilter...though I worry a bit about this change growing larger and larger

Sure, a small follow-up pr would probably be better.

Reply:
The change to make `HeaderFilter` mirror `ExcludeHeaderFilter` was not too bad, so I went ahead with it. I thought it might actually be better to do now while everything is fresh in my mind. 🙂